cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py

```diff
@@ -96,3 +109,81 @@ def test_provision_cloud_init(appliance, request, setup_provider, provider, prov
     with ssh.SSHClient(hostname=connect_ip, username=provisioning['ci-username'],
                        password=provisioning['ci-pass']) as ssh_client:
         wait_for(ssh_client.uptime, num_sec=200, handle_exception=True)
+
+
+@pytest.mark.rhv3
+@pytest.mark.uncollectif(lambda provider, appliance:
+    not provider.one_of(RHEVMProvider) or appliance.version < '5.10.0.12',
+    reason='Required customization template is only available in CFME 5.10.0.12 or greater.')
+def test_provision_cloud_init_payload(appliance, request, setup_provider, provider, provisioning,
+                                      vm_name):
+    """
+    Tests that options specified in VM provisioning dialog in UI are properly passed as a cloud-init
+    payload to the newly provisioned VM.
+
+    Metadata:
+        test_flag: cloud_init, provision
+    """
+    image = provisioning.get('ci-image', None)
+    if not image:
+        pytest.skip('No ci-image found in provider specification.')
+    note = ('Testing provisioning from image {} to vm {} on provider {}'.format(
+        image, vm_name, provider.key))
+    logger.info(note)
+
+    ci_payload = {
+        'root_password': 'mysecret',
+        'address_mode': 'Static',
+        'hostname': 'cimachine',
+        'ip_address': '10.1.1.10',
+        'subnet_mask': '23',
+        'gateway': '10.1.1.254',
+        'dns_servers': '10.1.1.200',
+        'dns_suffixes': 'virt.lab.example.com',
+        'custom_template': {'name': 'oVirt cloud-init'}
+    }
+
+    inst_args = {
+        'request': {'notes': note},
+        'customize': {'customize_type': 'Specification'},
+        'template_name': image
+    }
+
+    inst_args['customize'].update(ci_payload)
+    logger.info('Instance args: {}'.format(inst_args))
+
+    # Provision VM
+    collection = appliance.provider_based_collection(provider)
+    instance = collection.create(vm_name, provider, form_values=inst_args)
+    request.addfinalizer(instance.cleanup_on_provider)
+    provision_request = provider.appliance.collections.requests.instantiate(vm_name,
+                                                                            partial_check=True)
+    provision_request.wait_for_request()
+
+    connect_ip = wait_for(find_global_ipv6, func_args=[instance], num_sec=600, delay=20).out
+    logger.info('Connect IP: {}'.format(connect_ip))
+
+    # Connect to the newly provisioned VM
+    ssh_client = paramiko.SSHClient()
```

Comment:
Please use `cfme.utils.ssh.SSHClient`.  It inherits Paramiko, and supports using the client in a context manager to close and control transport connection between exec calls.

Use that context manager below to wrap the use of the client to call `run_command`, and remove the call to close being registered with `request.addfinalizer`

Reply:
Hi @mshriver, this is something I've mentioned in the description of the PR - for some reason, I was not able to get `cfme.utils.ssh.SSHClient` to work with IPv6.

Reply:
Gotcha, missed that in your description, my bad. Let's look into why its not working with IPv6, as it should just provide some convenience methods for Paramiko's library.

Reply:
@mshriver I think I know where the problem with wrapped paramiko might be. Working on a fix.